doc/perf.md

```diff
@@ -190,16 +183,15 @@ It is often easier to work with optimizations disabled (or with `-O1`),
 for two reasons:
 1. Optimizing a large program is slow. The optimized program will run faster,
     but it takes a while for LLVM to figure out how to make the code faster.
-    Waiting for the optimizer to do its thing is annoying.
-2. The optimizer assumes that your code doesn't do some dumb things.
-    This can have surprising consequences.
+    Waiting for the optimizer to do its thing is annoying. See Example #1
+2. The optimizer assumes that your code doesn't contain dumb things, that cause **undefined behavior** (see below). If it does it may behave unpredictably (not crashing when it should do so, etc.). See Example #2
 
 Let's explore these with more examples.
 
 
 ### Optimizing a large program is slow
```

Comment:
Maybe these titles should contain `#1` / `#2` in them. For example: `### Example #1: Optimizing a large program is slow`

Reply:
very well :D

Reply:
```suggestion
### Example #1: Optimizing a large program is slow
```